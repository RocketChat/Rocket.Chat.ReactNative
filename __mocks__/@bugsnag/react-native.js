export default {
	start: () => '',
	leaveBreadcrumb: () => '',
	notify: () => ''
};
