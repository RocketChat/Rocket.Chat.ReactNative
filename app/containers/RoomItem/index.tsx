import React, { useEffect, useReducer, useRef } from 'react';
import { Subscription } from 'rxjs';

import I18n from '../../i18n';
import { getUserPresence } from '../../lib/methods';
import { isGroupChat } from '../../lib/methods/helpers';
import { formatDate } from '../../lib/methods/helpers/room';
import { IRoomItemContainerProps } from './interfaces';
import RoomItem from './RoomItem';
import { ROW_HEIGHT, ROW_HEIGHT_CONDENSED } from './styles';
import { useUserStatus } from './useUserStatus';

export { ROW_HEIGHT, ROW_HEIGHT_CONDENSED };

const attrs = ['width', 'isFocused', 'showLastMessage', 'autoJoin', 'showAvatar', 'displayMode'];

const RoomItemContainer = React.memo(
	({
		item,
		id,
		onPress,
		onLongPress,
		width,
		toggleFav,
		toggleRead,
		hideChannel,
		isFocused,
		showLastMessage,
		username,
		useRealName,
		autoJoin,
		showAvatar,
		displayMode,
		getRoomTitle = () => 'title',
		getRoomAvatar = () => '',
		getIsRead = () => false,
		swipeEnabled = true
	}: IRoomItemContainerProps) => {
		const name = getRoomTitle(item);
		const testID = `rooms-list-view-item-${name}`;
		const avatar = getRoomAvatar(item);
		const isRead = getIsRead(item);
		const date = item.roomUpdatedAt && formatDate(item.roomUpdatedAt);
		const alert = item.alert || item.tunread?.length;
		const [_, forceUpdate] = useReducer(x => x + 1, 1);
		const roomSubscription = useRef<Subscription | null>(null);

		const { connected, status } = useUserStatus(item.t, item?.visitor?.status, id);

		useEffect(() => {
			const init = () => {
				if (item?.observe) {
					const observable = item.observe();
					roomSubscription.current = observable?.subscribe?.(() => {
						if (_) forceUpdate();
					});
				}
			};
			init();

			return () => roomSubscription.current?.unsubscribe();
		}, []);

		useEffect(() => {
			const isDirect = !!(item.t === 'd' && id && !isGroupChat(item));
			if (connected && isDirect) {
				getUserPresence(id);
			}
		}, [connected]);

		const handleOnPress = () => onPress(item);

		const handleOnLongPress = () => onLongPress && onLongPress(item);

		let accessibilityLabel = '';
		if (item.unread === 1) {
			accessibilityLabel = `, ${item.unread} ${I18n.t('alert')}`;
		} else if (item.unread > 1) {
			accessibilityLabel = `, ${item.unread} ${I18n.t('alerts')}`;
		}
		if (item.userMentions > 0) {
			accessibilityLabel = `, ${I18n.t('you_were_mentioned')}`;
		}
		if (date) {
			accessibilityLabel = `, ${I18n.t('last_message')} ${date}`;
		}

		return (
			<RoomItem
				name={name}
				avatar={avatar}
				isGroupChat={isGroupChat(item)}
				isRead={isRead}
				onPress={handleOnPress}
				onLongPress={handleOnLongPress}
				date={date}
				accessibilityLabel={accessibilityLabel}
				width={width}
				favorite={item.f}
				rid={item.rid}
				toggleFav={toggleFav}
				toggleRead={toggleRead}
				hideChannel={hideChannel}
				testID={testID}
				type={item.t}
				isFocused={isFocused}
				prid={item.prid}
				status={status}
				hideUnreadStatus={item.hideUnreadStatus}
				hideMentionStatus={item.hideMentionStatus}
				alert={alert}
				lastMessage={item.lastMessage}
				showLastMessage={showLastMessage}
				username={username}
				useRealName={useRealName}
				unread={item.unread}
				userMentions={item.userMentions}
				groupMentions={item.groupMentions}
				tunread={item.tunread}
				tunreadUser={item.tunreadUser}
				tunreadGroup={item.tunreadGroup}
				swipeEnabled={swipeEnabled}
				teamMain={item.teamMain}
				autoJoin={autoJoin}
				showAvatar={showAvatar}
				displayMode={displayMode}
				sourceType={item.source}
			/>
		);
	},
	(props, nextProps) => attrs.every(key => props[key] === nextProps[key])
);

export default RoomItemContainer;
