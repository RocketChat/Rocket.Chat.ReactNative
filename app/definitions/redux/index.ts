// ACTIONS
import { TActionInquiry } from '../../ee/omnichannel/actions/inquiry';
import { TActionActiveUsers } from '../../actions/activeUsers';
import { TActionApp } from '../../actions/app';
import { TActionCreateChannel } from '../../actions/createChannel';
import { TActionCreateDiscussion } from '../../actions/createDiscussion';
import { TActionCustomEmojis } from '../../actions/customEmojis';
import { TActionEncryption } from '../../actions/encryption';
import { TActionInviteLinks } from '../../actions/inviteLinks';
import { IActionRoles } from '../../actions/roles';
import { TActionSelectedUsers } from '../../actions/selectedUsers';
import { TActionServer } from '../../actions/server';
import { IActionSettings } from '../../actions/settings';
import { TActionsShare } from '../../actions/share';
import { TActionSortPreferences } from '../../actions/sortPreferences';
import { TActionUserTyping } from '../../actions/usersTyping';
import { TActionPermissions } from '../../actions/permissions';
import { TActionEnterpriseModules } from '../../actions/enterpriseModules';
// REDUCERS
import { IActiveUsers } from '../../reducers/activeUsers';
import { IApp } from '../../reducers/app';
import { IConnect } from '../../reducers/connect';
import { ICreateChannel } from '../../reducers/createChannel';
import { ICreateDiscussion } from '../../reducers/createDiscussion';
import { IEncryption } from '../../reducers/encryption';
import { IInviteLinks } from '../../reducers/inviteLinks';
import { ILogin } from '../../reducers/login';
import { IRoles } from '../../reducers/roles';
import { IRoom } from '../../reducers/room';
import { ISelectedUsers } from '../../reducers/selectedUsers';
import { IServer } from '../../reducers/server';
import { TSettingsState } from '../../reducers/settings';
import { IShare } from '../../reducers/share';
import { IInquiry } from '../../ee/omnichannel/reducers/inquiry';
import { IPermissionsState } from '../../reducers/permissions';
import { IEnterpriseModules } from '../../reducers/enterpriseModules';

export interface IApplicationState {
	settings: TSettingsState;
	login: ILogin;
	meteor: IConnect;
	server: IServer;
	selectedUsers: ISelectedUsers;
	app: IApp;
	createChannel: ICreateChannel;
	room: IRoom;
	rooms: any;
	sortPreferences: any;
	share: IShare;
	customEmojis: any;
	activeUsers: IActiveUsers;
	usersTyping: any;
	inviteLinks: IInviteLinks;
	createDiscussion: ICreateDiscussion;
	inquiry: IInquiry;
	enterpriseModules: IEnterpriseModules;
	encryption: IEncryption;
	permissions: IPermissionsState;
	roles: IRoles;
}

export type TApplicationActions = TActionActiveUsers &
	TActionSelectedUsers &
	TActionCustomEmojis &
	TActionInviteLinks &
	IActionRoles &
	IActionSettings &
	TActionEncryption &
	TActionSortPreferences &
	TActionUserTyping &
	TActionCreateDiscussion &
	TActionCreateChannel &
	TActionsShare &
	TActionServer &
	TActionApp &
	TActionInquiry &
	TActionPermissions &
	TActionEnterpriseModules;
