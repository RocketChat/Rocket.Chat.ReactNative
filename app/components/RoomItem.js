import React from 'react';
import { CachedImage } from 'react-native-img-cache';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import PropTypes from 'prop-types';
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native';

import avatarInitialsAndColor from '../utils/avatarInitialsAndColor';

const styles = StyleSheet.create({
	container: {
		// flex: 1,
		flexDirection: 'row',
		paddingLeft: 16,
		paddingRight: 16,
		height: 56,
		alignItems: 'center'
	},
	number: {
		minWidth: 20,
		borderRadius: 5,
		backgroundColor: '#1d74f5',
		color: '#fff',
		textAlign: 'center',
		overflow: 'hidden',
		fontSize: 14,
		paddingLeft: 5,
		paddingRight: 5
	},
	roomName: {
		flex: 1,
		fontSize: 16,
		color: '#444',
		marginLeft: 16,
		marginRight: 4
	},
	iconContainer: {
		height: 40,
		width: 40,
		borderRadius: 20,
		overflow: 'hidden',
		justifyContent: 'center',
		alignItems: 'center'
	},
	icon: {
		fontSize: 20,
		color: '#fff'
	},
	avatar: {
		width: 40,
		height: 40,
		position: 'absolute',
		borderRadius: 20
	},
	avatarInitials: {
		fontSize: 20,
		color: '#ffffff'
	}
});

export default class RoomItem extends React.PureComponent {
	static propTypes = {
		type: PropTypes.string.isRequired,
		name: PropTypes.string.isRequired,
		unread: PropTypes.number,
		baseUrl: PropTypes.string,
		onPress: PropTypes.func
	}

	get icon() {
		const { type, name, baseUrl } = this.props;

		const icon = {
			d: 'at',
			c: 'pound',
			p: 'lock',
			l: 'account'
		}[type];

		if (!icon) {
			return null;
		}

		if (type === 'd') {
			const { initials, color } = avatarInitialsAndColor(name);
			return (
				<View style={[styles.iconContainer, { backgroundColor: color }]}>
					<Text style={styles.avatarInitials}>{initials}</Text>
					<CachedImage style={styles.avatar} source={{ uri: `${ baseUrl }/avatar/${ name }` }} />
				</View>
			);
		}

		const { color } = avatarInitialsAndColor(name);

		return (
			<View style={[styles.iconContainer, { backgroundColor: color }]}>
				<MaterialCommunityIcons name={icon} style={styles.icon} />
			</View>
		);
	}

	renderNumber = (unread) => {
		if (!unread || unread <= 0) {
			return;
		}

		if (unread >= 1000) {
			unread = '999+';
		}

		return (
			<Text style={styles.number}>
				{ unread }
			</Text>
		);
	}

	render() {
		const { unread, name } = this.props;
		return (
			<TouchableOpacity onPress={this.props.onPress} style={styles.container}>
				{this.icon}
				<Text style={styles.roomName} ellipsizeMode='tail' numberOfLines={1}>{ name }</Text>
				{this.renderNumber(unread)}
			</TouchableOpacity>
		);
	}
}
