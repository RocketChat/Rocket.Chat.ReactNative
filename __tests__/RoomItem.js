import {View} from 'react-native';
import { Provider } from 'react-redux';

import { createStore, combineReducers } from 'redux';

const reducers = combineReducers({login:() => ({user: {}}), settings:() => ({}), meteor: () => ({ connected: true })});
const store = createStore(reducers);

import React from 'react';
import RoomItem from '../app/presentation/RoomItem';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const date = '2017-10-10T10:00:00Z';
const onPress = () => {};

it('renders correctly', () => {
	expect(renderer.create(<Provider store={store}><View><RoomItem onPress={onPress} type="d" _updatedAt={date} name="name" baseUrl="baseUrl" /></View></Provider>).toJSON()).toMatchSnapshot();
});

it('render unread', () => {
	expect(renderer.create(<Provider store={store}><View><RoomItem onPress={onPress} type="d" _updatedAt={date} name="name" unread={1} /></View></Provider>).toJSON()).toMatchSnapshot();
});

it('render unread +999', () => {
	expect(renderer.create(<Provider store={store}><View><RoomItem onPress={onPress} type="d" _updatedAt={date} name="name" unread={1000} /></View></Provider>).toJSON()).toMatchSnapshot();
});

it('render no icon', () => {
	expect(renderer.create(<Provider store={store}><View><RoomItem onPress={onPress} type="X" _updatedAt={date} name="name" /></View></Provider>).toJSON()).toMatchSnapshot();
});

it('render private group', () => {
	expect(renderer.create(<Provider store={store}><View><RoomItem onPress={onPress} type="g" _updatedAt={date} name="private-group" /> </View></Provider>).toJSON()).toMatchSnapshot();
});

it('render channel', () => {
	expect(renderer.create(<Provider store={store}><View><RoomItem onPress={onPress} type="c" _updatedAt={date} name="general" /></View></Provider>).toJSON()).toMatchSnapshot();
});
