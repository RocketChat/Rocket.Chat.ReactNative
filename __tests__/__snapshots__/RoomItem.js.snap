// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render channel 1`] = `
<View>
  <View
    accessibilityLabel="general, last message Nov 10"
    accessibilityTraits="selected"
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomColor": "#ddd",
            "borderBottomWidth": 0.5,
            "flexDirection": "row",
            "paddingHorizontal": 16,
            "paddingVertical": 12,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "backgroundColor": "#00BCD4",
              "borderRadius": 2,
              "height": 46,
              "width": 46,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#ffffff",
              },
              Object {
                "fontSize": 28.75,
                "fontWeight": "800",
              },
            ]
          }
        >
          G
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "height": "100%",
            "marginLeft": 16,
            "marginRight": 4,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 15,
                },
                Object {
                  "marginRight": 5,
                  "marginTop": 3,
                },
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            Ôê£
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#444",
                  "flex": 1,
                  "fontSize": 18,
                  "marginRight": 8,
                },
                undefined,
              ]
            }
          >
            general
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "color": "#888",
                  "fontSize": 10,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            Nov 10
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`render no icon 1`] = `
<View>
  <View
    accessibilityLabel="name, last message Nov 10"
    accessibilityTraits="selected"
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomColor": "#ddd",
            "borderBottomWidth": 0.5,
            "flexDirection": "row",
            "paddingHorizontal": 16,
            "paddingVertical": 12,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "backgroundColor": "#3F51B5",
              "borderRadius": 2,
              "height": 46,
              "width": 46,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#ffffff",
              },
              Object {
                "fontSize": 28.75,
                "fontWeight": "800",
              },
            ]
          }
        >
          N
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "height": "100%",
            "marginLeft": 16,
            "marginRight": 4,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 15,
                },
                Object {
                  "marginRight": 5,
                  "marginTop": 3,
                },
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#444",
                  "flex": 1,
                  "fontSize": 18,
                  "marginRight": 8,
                },
                undefined,
              ]
            }
          >
            name
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "color": "#888",
                  "fontSize": 10,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            Nov 10
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`render private group 1`] = `
<View>
  <View
    accessibilityLabel="private-group, last message Nov 10"
    accessibilityTraits="selected"
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomColor": "#ddd",
            "borderBottomWidth": 0.5,
            "flexDirection": "row",
            "paddingHorizontal": 16,
            "paddingVertical": 12,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "backgroundColor": "#FF9800",
              "borderRadius": 2,
              "height": 46,
              "width": 46,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#ffffff",
              },
              Object {
                "fontSize": 28.75,
                "fontWeight": "800",
              },
            ]
          }
        >
          P
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "height": "100%",
            "marginLeft": 16,
            "marginRight": 4,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 15,
                },
                Object {
                  "marginRight": 5,
                  "marginTop": 3,
                },
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#444",
                  "flex": 1,
                  "fontSize": 18,
                  "marginRight": 8,
                },
                undefined,
              ]
            }
          >
            private-group
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "color": "#888",
                  "fontSize": 10,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            Nov 10
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        />
      </View>
    </View>
  </View>
   
</View>
`;

exports[`render unread +999 1`] = `
<View>
  <View
    accessibilityLabel="name, 1000 alerts, last message Nov 10"
    accessibilityTraits="selected"
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomColor": "#ddd",
            "borderBottomWidth": 0.5,
            "flexDirection": "row",
            "paddingHorizontal": 16,
            "paddingVertical": 12,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "backgroundColor": "#3F51B5",
              "borderRadius": 2,
              "height": 46,
              "width": 46,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#ffffff",
              },
              Object {
                "fontSize": 28.75,
                "fontWeight": "800",
              },
            ]
          }
        >
          N
        </Text>
        <View
          style={
            Array [
              Array [
                Object {
                  "position": "absolute",
                },
                Object {
                  "borderRadius": 2,
                  "height": 46,
                  "width": 46,
                },
              ],
              Object {
                "overflow": "hidden",
              },
            ]
          }
        >
          <FastImageView
            resizeMode="cover"
            source={
              Object {
                "priority": "high",
                "uri": "/avatar/name",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "borderRadius": 16,
                "height": 16,
                "width": 16,
              },
              Array [
                Object {
                  "borderColor": "#fff",
                  "borderWidth": 3,
                  "bottom": -3,
                  "position": "absolute",
                  "right": -3,
                },
                undefined,
              ],
              Object {
                "backgroundColor": "#cbced1",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "height": "100%",
            "marginLeft": 16,
            "marginRight": 4,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 15,
                },
                Object {
                  "marginRight": 5,
                  "marginTop": 3,
                },
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            ÔÅ•
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#444",
                  "flex": 1,
                  "fontSize": 18,
                  "marginRight": 8,
                },
                undefined,
              ]
            }
          >
            name
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "color": "#888",
                  "fontSize": 10,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            Nov 10
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#1d74f5",
                "borderRadius": 4,
                "color": "#fff",
                "fontSize": 14,
                "justifyContent": "center",
                "minWidth": 25,
                "overflow": "hidden",
                "paddingHorizontal": 5,
                "paddingVertical": 4,
                "textAlign": "center",
              }
            }
          >
            999+
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`render unread 1`] = `
<View>
  <View
    accessibilityLabel="name, 1 alert, last message Nov 10"
    accessibilityTraits="selected"
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomColor": "#ddd",
            "borderBottomWidth": 0.5,
            "flexDirection": "row",
            "paddingHorizontal": 16,
            "paddingVertical": 12,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "backgroundColor": "#3F51B5",
              "borderRadius": 2,
              "height": 46,
              "width": 46,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#ffffff",
              },
              Object {
                "fontSize": 28.75,
                "fontWeight": "800",
              },
            ]
          }
        >
          N
        </Text>
        <View
          style={
            Array [
              Array [
                Object {
                  "position": "absolute",
                },
                Object {
                  "borderRadius": 2,
                  "height": 46,
                  "width": 46,
                },
              ],
              Object {
                "overflow": "hidden",
              },
            ]
          }
        >
          <FastImageView
            resizeMode="cover"
            source={
              Object {
                "priority": "high",
                "uri": "/avatar/name",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "borderRadius": 16,
                "height": 16,
                "width": 16,
              },
              Array [
                Object {
                  "borderColor": "#fff",
                  "borderWidth": 3,
                  "bottom": -3,
                  "position": "absolute",
                  "right": -3,
                },
                undefined,
              ],
              Object {
                "backgroundColor": "#cbced1",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "height": "100%",
            "marginLeft": 16,
            "marginRight": 4,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 15,
                },
                Object {
                  "marginRight": 5,
                  "marginTop": 3,
                },
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            ÔÅ•
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#444",
                  "flex": 1,
                  "fontSize": 18,
                  "marginRight": 8,
                },
                undefined,
              ]
            }
          >
            name
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "color": "#888",
                  "fontSize": 10,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            Nov 10
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#1d74f5",
                "borderRadius": 4,
                "color": "#fff",
                "fontSize": 14,
                "justifyContent": "center",
                "minWidth": 25,
                "overflow": "hidden",
                "paddingHorizontal": 5,
                "paddingVertical": 4,
                "textAlign": "center",
              }
            }
          >
            1
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View>
  <View
    accessibilityLabel="name, last message Nov 10"
    accessibilityTraits="selected"
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomColor": "#ddd",
            "borderBottomWidth": 0.5,
            "flexDirection": "row",
            "paddingHorizontal": 16,
            "paddingVertical": 12,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "backgroundColor": "#3F51B5",
              "borderRadius": 2,
              "height": 46,
              "width": 46,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#ffffff",
              },
              Object {
                "fontSize": 28.75,
                "fontWeight": "800",
              },
            ]
          }
        >
          N
        </Text>
        <View
          style={
            Array [
              Array [
                Object {
                  "position": "absolute",
                },
                Object {
                  "borderRadius": 2,
                  "height": 46,
                  "width": 46,
                },
              ],
              Object {
                "overflow": "hidden",
              },
            ]
          }
        >
          <FastImageView
            resizeMode="cover"
            source={
              Object {
                "priority": "high",
                "uri": "/avatar/name",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "borderRadius": 16,
                "height": 16,
                "width": 16,
              },
              Array [
                Object {
                  "borderColor": "#fff",
                  "borderWidth": 3,
                  "bottom": -3,
                  "position": "absolute",
                  "right": -3,
                },
                undefined,
              ],
              Object {
                "backgroundColor": "#cbced1",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "height": "100%",
            "marginLeft": 16,
            "marginRight": 4,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 15,
                },
                Object {
                  "marginRight": 5,
                  "marginTop": 3,
                },
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            ÔÅ•
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#444",
                  "flex": 1,
                  "fontSize": 18,
                  "marginRight": 8,
                },
                undefined,
              ]
            }
          >
            name
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "color": "#888",
                  "fontSize": 10,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            Nov 10
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
